src/main/java/com/xero/api/XeroClient.java

```diff
@@ -219,6 +150,66 @@ protected Response put(String endPoint, JAXBElement<?> object)  {
 		}
 	}
 
+	protected Response put(String endPoint, String contentType, byte[] bytes) throws IOException {
+		HttpResponse resp = null;
+		OAuthRequestResource req = new OAuthRequestResource(config, endPoint,"PUT", contentType, bytes,null);
+		req.setToken(token);
+		req.setTokenSecret(tokenSecret);
+
+		try {
+			resp = req.execute();
+			return unmarshallResponse(resp.parseAsString(), Response.class);
+		} catch (IOException ioe) {
+			if (ioe instanceof HttpResponseException) {
+				HttpResponseException googleException = (HttpResponseException)ioe;
+
+				if (googleException.getStatusCode() == 400 ||
+						googleException.getStatusCode() == 401 ||
+						googleException.getStatusCode() == 404 ||
+						googleException.getStatusCode() == 500 ||
+						googleException.getStatusCode() == 503) {
+					throw newApiException(googleException);
+				} else {
+					System.out.println("Error - not tested with newApiException method");
```

Comment:
You should consider using a logging framework such as SLF4J instead of printing directly to the console.